dashboard/app/models/user.rb

```diff
@@ -848,7 +855,7 @@ def self.find_for_authentication(tainted_conditions)
       ).first
     elsif hashed_email = conditions.delete(:hashed_email)
       return nil if hashed_email.utf8mb4?
-      where(hashed_email: hashed_email).first
+      return find_by_hashed_email(hashed_email)
```

Comment:
I wonder if that `nil if hashed_email.utf8mb4?` check ought to live inside `find_by_hashed_email` in the model?